app/controllers/registers_controller.rb

```diff
@@ -57,36 +57,64 @@ def show
 
   def history
     @register = Spina::Register.find_by_slug!(params[:id])
-    @register_data = @registers_client.get_register(@register.name.parameterize, @register.register_phase)
-    fields = @register_data.get_field_definitions.map { |field| field.item.value['field'] }
-    all_records = @register_data.get_records_with_history
-    entries_reversed = @register_data.get_entries.reverse_each
 
-    entries_mapped_with_items = entries_reversed.map do |entry|
-      records_for_key = all_records.get_records_for_key(entry.key)
-      current_record = records_for_key.detect { |record| record.entry.entry_number == entry.entry_number }
-      previous_record_index = records_for_key.find_index(current_record) - 1
+    unless @register.populated
+      populate_register(@register)
+    end
+
+    entries = recover_entries_history(@register.id, @register.fields, params)
+    register_data = @registers_client.get_register(@register.name.parameterize, @register.register_phase)
+    fields = register_data.get_field_definitions.map { |field| field.item.value['field'] }
```

Comment:
Could this use the `get_field_definitions` method in this controller, rather than querying the client library?

Reply:
It probably could if that's the way we want to go.